pkg/command/buildbase.go

```diff
@@ -71,6 +88,21 @@ func (opts *BaseBuildOptions) Validate(cmd *cobra.Command, args []string) error
 	return nil
 }
 
-func (opts *BaseBuildOptions) tag() (name.Tag, error) {
-	return name.NewTag(opts.ImageName, name.WeakValidation)
+type imageNameContext struct {
+	url.URL
+}
+
+var imageNameFunctions = template.FuncMap{
```

Comment:
This is really cool.

Reply:
yeah, go templates are stupidly powerful 😈